afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/cmap.c

```diff
@@ -719,64 +719,125 @@ static Format2 *makeFormat2(cmapCtx h) {
 
 /* ------- end format 2 */
 
+static void setOrdered( Mapping *mapping, int codeBreak, int span, int index)
+{
+    int threshold;
+    /* If the ordered segment starts with a code break or ends with a code break,
+     then the cost of keeping this as a separate segment is just one new segment break
+     (the following segment break).
+     Else, the cost is adding two new segment breaks - the one to start this segment,
+     and to start the next segment. We mark as NOT orderd those segments that are no larger
+     than the threshold.
+     */
+    if ( (mapping[index-span].flags & CODE_BREAK) || codeBreak)
+        threshold = 4;
+    else
+        threshold = 8;
+    if (span > threshold)
+        mapping[index - span].ordered = 1;
+
+}
+
 /* Partition mappings optimally for format 4. Assumes mapping in code order */
 static int partitionRanges(cmapCtx h, Mapping *mapping) {
 	int span;
 	int ordered;
-	int overflow;
+    int lastOrdered;
 	int total;
 	int i;
-	int nSegments;
-
-	/* Create ordered partitioning */
+    int nSegments;
+
+    /* mapping is sorted by charCode, then by glyphID.
+     If there is a break on the character code sequence, we always make a new segment.
+     If there is a break in the glyph ID sequence, then we may make a segment break, if doing so saves more bytes
+     than not doing so.
+     If a segment has more than one glyph and no breaks in its glyphID sequence, then it is set as  ordered.
+     Ordered segments are dropped only if keeping them costs less than the new segment cost.
+     All other segments are merged togther.
+     idRangeOffset for an ordered segment is set to 0, and its glyphID's are not added to the glyphId array,
+     as its glyphIDs can be calculated from startCode/endCode/idDelta alone.
+     */
+    
+    /* Create ordered partitioning. */
 	span = 1;
 	for (i = 1; i < h->mapping.cnt; i++) {
 		int codeBreak = (mapping[i - 1].code + 1 != mapping[i].code);
 		if (codeBreak || mapping[i - 1].glyphId + 1 != mapping[i].glyphId) {
-			mapping[i - span].span = span;
-			if (codeBreak) {
-				mapping[i].flags |= CODE_BREAK;
-			}
+            mapping[i - span].span = span;
+            mapping[i - span].ordered = 0; /* default */
+            if (span > 1) /* an ordered segment is a segment with more than one code point with consecutive glyph ID's */
+            {
+                setOrdered(mapping, codeBreak, span, i);
+            }
+            if (codeBreak) {
+                mapping[i].flags |= CODE_BREAK;
+            }
 			span = 1;
+            nSegments++;
```

Comment:
seems to be another mix of tabs vs spaces here, please make the indentation consistent